### PR TITLE
fix: Import failing when application theme is set but deleted

### LINKED ISSUE #25032: [Bug]: Discard fails with unable to import application error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2023-07-03 at 5 39 23 PM

### Steps To Reproduce

1. Go to this [app](https://ce-24764.dp.appsmith.com/app/canvas-splitting-test/page1-649b1ed5e5403e34f502ea11/edit?branch=master)
2. Open git sync modal and discard
3. Observe error is thrown

### Public Sample App

_No response_

### Environment

Deploy Preview

### Issue video log

_No response_

### Version

Deploy preview

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
The import process fails if an application has a deleted theme. This PR fixes this issue.

#### PR fixes following issue(s)
Fixes #25032

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [x] Jest
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by AnaghHegde — COMMENTED** [review]
Minor things, rest LGTM

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ThemeServiceTest.java [hunk 1]
```diff
@@ -800,4 +800,44 @@ public void importThemesToApplication_WhenBothImportedThemesAreCustom_NewThemesC
...
+        Mono<Application> applicationMono = Mono.just(createApplication())
+                .map(application -> {
+                    // setting invalid ids to themes to check the case
+                    application.setEditModeThemeId(UUID.randomUUID().toString());
+                    application.setPublishedModeThemeId(UUID.randomUUID().toString());
+                    return application;
+                })
+                .flatMap(applicationRepository::save)
+                .flatMap(savedApplication -> {
+                    assert savedApplication.getId() != null;
```
[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ThemeServiceTest.java:827]: It's not required. The IDE was suggesting this and I accepted as it's a harmless operation.

**Review by AnaghHegde — APPROVED** [review]

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**